services/staging/rs_server_staging/processors.py

```diff
@@ -11,22 +11,950 @@
+    STARTED = "started"  # Processor execution has started
+    IN_PROGRESS = "in_progress"  # Processor execution is in progress
+    STOPPED = "stopped"
+    FAILED = "failed"
+    FINISHED = "finished"
+    PAUSED = "paused"
+    RESUMED = "resumed"
+    CANCELLED = "cancelled"
+
+    # Serialization
+    def __str__(self):
+        """
+        Returns the string representation of the ProcessorStatus instance.
+
+        Returns:
+            str: The value of the current ProcessorStatus instance.
+        """
+        return self.value
+
+    @classmethod
+    def to_json(cls, status):
+        """
+        Serializes a ProcessorStatus instance to its JSON-compatible representation.
+        """
+        if isinstance(status, cls):
+            return status.value
+        raise ValueError("Invalid ProcessorStatus")
+
+    @classmethod
+    def from_json(cls, value):
+        """
+        Deserializes a string value back into a ProcessorStatus instance.
+        """
+        for status in cls:
+            if status.value == value:
+                return status
+        raise ValueError(f"Invalid ProcessorStatus value: {value}")
+
+
+# Custom authentication class
+class TokenAuth(AuthBase):
+    """Custom authentication class
+
+    Args:
+        AuthBase (ABC): Base auth class
+    """
+
+    def __init__(self, token: str):
+        """Init token auth
+
+        Args:
+            token (str): Token value
+        """
+        self.token = token
+
+    def __call__(self, request: Request):  # type: ignore
+        """Add the Authorization header to the request
+
+        Args:
+            request (Request): request to be modified
+
+        Returns:
+            Request: request with modified headers
+        """
+        request.headers = {  # type: ignore
+            "Authorization": f"Bearer {self.token}",
+            "Content-Type": "application/x-www-form-urlencoded",
+        }
+        return request
+
+    def __repr__(self) -> str:
+        return "RSPY Token handler"
+
+
+def streaming_task(product_url: str, auth: str, bucket: str, s3_file: str):
+    """
+    Streams a file from a product URL and uploads it to an S3-compatible storage.
+
+    This function downloads a file from the specified `product_url` using provided
+    authentication and uploads it to an S3 bucket using a streaming mechanism.
+    If no S3 handler is provided, it initializes a default `S3StorageHandler` using
+    environment variables for credentials.
+
+    Args:
+        product_url (str): The URL of the product to download.
+        auth (str): The authentication token or credentials required for the download.
+        s3_file (str): The destination path/key in the S3 bucket where the file will be uploaded.
+
+    Returns:
+        str: The S3 file path where the file was uploaded.
+
+    Raises:
+        ValueError: If the streaming process fails, raises a ValueError with details of the failure.
+    """
+
+    try:
```

Comment:
Maybe try to create this object only once, and send it as parameter.

Reply:
This is actually the code that runs in the dask cluster. What if other s3 env vars are used in that image ?